### PR TITLE
[FIX] Incorrect display of "Reply in Direct Message" in MessageAction

### PR DESCRIPTION
**Pull request by abrom**
## Proposed changes
Check that we either already have a DM room started with the message user or that we have the `create-d` permission to start one, otherwise hiding the reply in DM menu option.

## Issue(s)
Currently if a user does not have the `create-d` permission and does not already have a DM started with a user, the "Reply in Direct Message" context menu option is still displayed for messages even though clicking it will show the user a "User not found - Invalid room" page and "Not allowed [error-not-allowed]" toast message.

## Screenshots
Screen Shot 2020-06-19 at 4 46 59 pm

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Changelog

[FIX] Incorrect display of "Reply in Direct Message" in MessageAction

## Further comments
Note I've left the tests in a separate commit for now as they weren't super simple due to some of the setup required to test this. I tried to find other examples of tests that did similar things but didn't really find anything suitable. Would be good to get some feedback as to if there would be a better way to do the setup?

### CODE REVIEW

File: app/ui-utils/client/lib/MessageAction.js [hunk 1]
```diff
@@ -172,13 +172,21 @@ Meteor.startup(async function() {
...
 			if (subscription == null) {
 				return false;
 			}
 			if (room.t === 'd' || room.t === 'l') {
 				return false;
 			}
+
+			// Check if we already have a DM started with the message user (not ourselves) or we can start one
+			const dmRoom = Rooms.findOne({ _id: [u._id, msg.u._id].sort().join('') });
+			const canAccessDM = dmRoom && Subscriptions.findOne({ rid: dmRoom._id, 'u._id': u._id });
```
Comment (MartinSchoeler) [inline app/ui-utils/client/lib/MessageAction.js:185]: Maybe before doing multiple `findOne`s, we can check if the user lack the permission, so we can avoid calling these methods when we don't need it.

Comment (abrom) [reply app/ui-utils/client/lib/MessageAction.js:185]: Are you suggesting something like this?

```
// Don't show reply option if we're messaging someone else and we don't have
// permission to start a new DM with them
if (u._id !== msg.u._id && !hasPermission('create-d')) {
  const dmRoom = Rooms.findOne({ _id: [u._id, msg.u._id].sort().join('') });
  if (!dmRoom || !Subscriptions.findOne({ rid: dmRoom._id, 'u._id': u._id })) {
    return false;
  }
}
```

Comment (abrom) [reply app/ui-utils/client/lib/MessageAction.js:185]: @MartinSchoeler are you suggesting something like :point_up: ?

Comment (abrom) [reply app/ui-utils/client/lib/MessageAction.js:185]: @MartinSchoeler thoughts RE my suggested fix?

Comment (abrom) [reply app/ui-utils/client/lib/MessageAction.js:185]: hi @MartinSchoeler still awaiting your feedback.

**Review by MartinSchoeler — APPROVED** [review]
Yes, those changes are good! Sorry for the delay, I had some issues with my GH notifications

**Comment by abrom** [comment]
What's needed to get this moving along @MartinSchoeler ?

**Pull request merged**